apps/voting/src/pages/Home/Home.tsx

```diff
@@ -0,0 +1,191 @@
+    };
+
+    /**
+     * Opens a confirmation modal for ballot submission. If not connected at this point, a connection prompt will be
+     * made prior to opening the modal.
+     */
+    const submit = useCallback(() => {
+        if ((wallet?.connection === undefined || wallet?.account === undefined) && openViewConnection !== undefined) {
+            openViewConnection();
+            setAwaitConnection(true);
+        } else {
+            setConfirmOpen(true);
+        }
+    }, [wallet?.connection, openViewConnection, wallet?.account]);
+
+    // Handle connecting due to a submission attempt while not connected.
+    useEffect(() => {
+        if (awaitConnection && wallet?.connection !== undefined && wallet?.account !== undefined) {
+            submit();
+            setAwaitConnection(false);
+        }
+    }, [awaitConnection, wallet?.connection, submit, wallet?.account]);
+
+    if (electionConfig === undefined) {
+        return null;
+    }
+
+    return (
+        <>
+            <Row className="mt-n4">
+                {electionConfig?.candidates.map((c) => (
+                    <Candidate
+                        key={c.index}
+                        candidate={c}
+                        onClick={() => toggleCandidate(c.index)}
+                        isSelected={selected.includes(c.index)}
+                    />
+                ))}
+            </Row>
+            {isElectionOpen && (
+                <div className="d-flex justify-content-center mt-4">
+                    <Button className="text-center" variant="primary" onClick={submit}>
+                        Submit
+                    </Button>
+                </div>
+            )}
+            <Modal show={confirmOpen} onHide={closeConfirm} backdrop="static">
+                <Modal.Header closeButton>
+                    <Modal.Title>Modal heading</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    {selected.length === 0 && (
+                        <p>
+                            You have not selected any candidates. Confirming this submission will result in en empty
+                            ballot.
+                        </p>
+                    )}
+                    {selected.length === electionConfig.candidates.length && (
+                        <p>
+                            You have selected all candidates. Confirming this submission will result in a ballot with
+                            equal weight distribution on all candidates, which is essentially the same as an empty
+                            ballot.
+                        </p>
+                    )}
+                    {0 < selected.length && selected.length < electionConfig.candidates.length && (
+                        <>
+                            <p>
+                                You have selected {selected.length} out of {electionConfig.candidates.length}{' '}
+                                candidates:
```

Comment:
Well, the candidates selected are also listed beneath this? I thought it would be nice as an added verification measure for the user, to have this. Would it make more sense to simply write "You have selected the following candidates:"?

Reply:
I would prefer that, yes.